### PR TITLE
Add ERReadmissionMIMIC4 task for ER-specific 30-day readmission prediction

### PR DESCRIPTION
**Pull request by RajendranRanjith**
## Contributor Information

- **Name:** Ranjithkumar Rajendran
- **NetID:** rr54
- **Email:** rr54@illinois.edu
- **Course:** CS598 DL4H — Spring 2026

## Contribution Type

**Standalone Task** — New task for MIMIC-IV ER-specific readmission prediction.

## Paper Reference

**KEEP: Integrating Medical Ontologies with Clinical Data for Robust Code Embeddings**
- Authors: Elhussein et al.
- Venue: CHIL 2025
- Link: https://arxiv.org/abs/2510.05049

## Description

This PR adds `ERReadmissionMIMIC4`, a new PyHealth task that predicts 30-day readmission for Emergency Room patients in MIMIC-IV. This implements the ER cohort definition from the KEEP paper, which demonstrated that ER patients have distinct readmission patterns requiring specialized prediction models.

### How it works:
1. Filters admissions where `admission_location == 'EMERGENCY ROOM'`
2. Extracts versioned ICD diagnosis codes (e.g., `"10_E11.9"`)
3. Computes discharge time from the admissions table
4. Labels each visit: `1` if next admission < 30 days, `0` otherwise
5. Returns PyHealth-compatible sample dicts with `conditions` and `readmission` fields

### Key design decisions:
- Inherits from `BaseTask` following PyHealth conventions
- Configurable readmission window via `timedelta` parameter (default: 30 days)
- Skips patients with < 2 admissions and visits with no diagnoses
- ICD codes versioned as `"{version}_{code}"` matching `ReadmissionPredictionMIMIC4`
- PEP8 compliant (88-char limit), Google-style docstrings, full type hints

## File Guide

| # | File | Description |
|---|------|-------------|
| 1 | `pyhealth/tasks/mimic4_er_readmission.py` | **Core task implementation** — `ERReadmissionMIMIC4` class (~130 lines) |
| 2 | `pyhealth/tasks/__init__.py` | Updated to export `ERReadmissionMIMIC4` |
| 3 | `tests/test_mimic4_er_readmission.py` | **7 unit tests** using synthetic data (edge cases: single admission, no diagnoses, custom window, ICD versioning) |
| 4 | `examples/mimic4_er_readmission_retain.py` | Ablation script: Standard vs ER-specific readmission using RETAIN |
| 5 | `examples/mimic4_er_readmission_transformer.py` | Ablation script: RETAIN vs Transformer on ER cohort |
| 6 | `docs/api/tasks/pyhealth.tasks.mimic4_er_readmission.rst` | Sphinx API documentation |
| 7 | `docs/api/tasks.rst` | Updated task index with new entry |

## Tests

All 7 unit tests pass using synthetic Patient objects (no real data required):

### CODE REVIEW

File: pyhealth/tasks/mimic4_er_readmission.py [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+    """ER-Specific Readmission prediction on MIMIC-IV.
+
+    Predicts whether an emergency-room patient will be
+    readmitted within a specified window (default 30 days)
+    based on clinical information from the current ER visit.
+
+    Only visits whose ``admission_location`` is
+    ``'EMERGENCY ROOM'`` are considered.  Diagnosis codes
+    are prefixed with their ICD version (``"9_"`` or
+    ``"10_"``) to match the format used by
```
Comment (jhnwu3) [inline pyhealth/tasks/mimic4_er_readmission.py:23]: https://pyhealth.readthedocs.io/en/latest/_modules/pyhealth/tasks/readmission_prediction.html#ReadmissionPredictionMIMIC3

Not clear to me how this is different than the already existing readmission prediction task lol.

**Review by jhnwu3 — CHANGES_REQUESTED** [review]

File: pyhealth/tasks/mimic4_er_readmission.py (hunk 1 above)
Comment (RajendranRanjith) [reply pyhealth/tasks/mimic4_er_readmission.py:23]: Thanks for reviewing the PR. While they are both readmission tasks, ERReadmissionMIMIC4 represents a distinct clinical benchmark (specifically mirroring the task defined in the KEEP CHIL 2025 paper) and differs from the general ReadmissionPredictionMIMIC4 in a few key ways:

**Target Population:** 
ERReadmissionMIMIC4 specifically filters for admission_location == 'EMERGENCY ROOM', whereas the existing task looks at all inpatient hospitalizations. ER readmission is a specific, widely studied clinical problem.

**Modality Requirements:** The existing ReadmissionPredictionMIMIC4 task requires every visit to have conditions, procedures, AND drugs (excluding visits that lack any of the three). ERReadmissionMIMIC4 only requires conditions (diagnoses), which is much more appropriate for ER visits where procedures or prescriptions aren't always comprehensively logged.

**Time Window:** The default window is 30 days (the standard for ER readmission benchmarks) instead of the 15-day default in the existing task.

I made sure to note the ER-specific admission_location filtering in the docstring, but let me know if you'd like me to add a more explicit comparison to the general readmission task in the documentation to make this clearer for future users!

Comment (RajendranRanjith) [reply pyhealth/tasks/mimic4_er_readmission.py:23]: @jhnwu3 I've clarified the differences in the thread above. Let me know if this makes sense
